### DISCUSSION TITLE
Expected MatanYadaev\EloquentSpatial\GeometryCast, string given

### DISCUSSION THREAD

**Discussion by RahulKRaj7** [oddvalue/laravel-drafts#10] [Q&A] [answered]
While using spatial field in model. i am facing `Expected MatanYadaev\EloquentSpatial\GeometryCast, string given` this type of error when trying to publish draft model.

The field is casted in model as 
`location' => Point::class,`

in the migration the field is created as
`$table->point('location')->nullable();`

**Comment by oddvalue** [comment, marked as answer]
Hey, this doesn't sound related to this package. If you remove the `HasDrafts` trait from the model do you still experience the error?

**Reply by RahulKRaj7** [reply]
Yes, it's not related to this package. i solved this issue by using mutators. Thanks for the reply.
**Edit:** This issue is related to [laravel-eloquent-spatial ](https://github.com/MatanYadaev/laravel-eloquent-spatial) package and it's solved here https://github.com/MatanYadaev/laravel-eloquent-spatial/issues/79
